gui/randogui.py

```diff
@@ -708,6 +786,255 @@ def launch_tricks_dialog(self):
+                        self.options.set_option("tunic-swap", False)
+                else:
+                    self.ui.option_tunic_swap.setEnabled(True)
+
+                self.save_settings()
+            case "Loftwing":
+                self.options.set_option(
+                    "selected-loftwing-model-pack", self.current_model_pack
+                )
+                self.save_settings()
+
+        self.update_model_customisation()
+
+    def read_metadata(self):
+        if self.current_model_pack == "Default":
+            self.metadata_path = (
+                CUSTOM_MODELS_PATH
+                / "Default"
+                / self.current_model_type
+                / "metadata.json"
+            )
+            if not os.path.isfile(self.metadata_path):
+                self.metadata_path.write_bytes(
+                    (
+                        LINK_MODEL_DATA_PATH
+                        / self.current_model_type
+                        / "defaultMetadata.json"
+                    ).read_bytes()
+                )
+        else:
+            self.metadata_path = (
+                CUSTOM_MODELS_PATH
+                / self.current_model_pack
+                / self.current_model_type
+                / "metadata.json"
+            )
+
+        if os.path.isfile(self.metadata_path):
+            with open(self.metadata_path) as f:
+                self.metadata = json.load(f)
+        else:
+            self.metadata = {}
+
+    def model_color_changed(self, color: str, name: str):
+        self.metadata["Colors"][name] = color
+
+        with open(self.metadata_path, "w") as f:
+            json.dump(self.metadata, f)
+
+        self.update_model_preview()
+
+    def update_model_customisation(self):
+        color_box_index = 0
+
+        while (
+            self.color_box.count() > 2
+        ):  # leaves horizonal and vertical spacers in box
+            layout = self.color_box.takeAt(0).layout()
+            while layout.count() > 0:
+                widget = layout.takeAt(0).widget()
+                widget.deleteLater()
+            layout.deleteLater()
+
+        self.color_buttons.clear()
+
+        author_layout = QVBoxLayout()
+        author_layout_index = 0
+
+        if author_name := self.metadata.get("ModelAuthorName"):
+            author_name_label = QLabel(f"Model Author: {author_name}")
+            author_name_label.setWordWrap(True)
+            author_layout.insertWidget(author_layout_index, author_name_label)
+            author_layout_index += 1
+
+        if author_comment := self.metadata.get("ModelAuthorComment"):
+            author_comment_label = QLabel(f"Model Author Comment: {author_comment}")
+            author_comment_label.setWordWrap(True)
+            author_layout.insertWidget(author_layout_index, author_comment_label)
+
+        self.color_box.insertLayout(color_box_index, author_layout)
+        color_box_index += 1
+
+        if colorData := self.metadata.get("Colors"):
```

Comment:
```suggestion
        if color_data := self.metadata.get("Colors"):
```
I think we should avoid mixing different casing styles right next to each other